ethcore/sync/src/chain/supplier.rs

```diff
@@ -145,14 +145,47 @@ impl SyncSupplier {
 				}
 			};
 
-			result.unwrap_or_else(|e| {
-				debug!(target:"sync", "{} -> Malformed packet {} : {}", peer, packet_id, e);
-			})
+			match result {
+				Err(PacketProcessError::Decoder(e)) => debug!(target:"sync", "{} -> Malformed packet {} : {}", peer, packet_id, e),
+				Err(PacketProcessError::ClientBusy) => sync.write().add_delayed_request(peer, packet_id, data),
+				Ok(()) => {}
+			}
+		}
+	}
+
+	/// Dispatch delayed requests
+	/// The main difference with dispatch packet is the direct send of the responses to the peer
+	pub fn postponed_dispatch_packet(sync: &RwLock<ChainSync>, io: &mut dyn SyncIo, peer: PeerId, packet_id: u8, data: &[u8]) {
+		let rlp = Rlp::new(data);
+
+		if let Some(id) = SyncPacket::from_u8(packet_id) {
+			let result = match id {
+				GetBlockHeadersPacket => SyncSupplier::send_rlp(
+					io, &rlp, peer,
+					SyncSupplier::return_block_headers,
+					|e| format!("Error sending block headers: {:?}", e)),
+
+				_ => {
+					debug!(target:"sync", "Unexpected packet {} was dispatched for delayed processing", packet_id);
+					Ok(())
+				}
```

Comment:
Is this case really possible?

Reply:
Right now, no indeed. But I've left this for future need, if some other packet is dispatched in postponed manner